components/cas-inbound-authenticator/src/main/java/org/wso2/carbon/identity/sso/cas/processor/SSOLoginProcessor.java

```diff
@@ -76,7 +73,31 @@ public boolean canHandle(IdentityRequest identityRequest) {
         CASResponse.CASResponseBuilder builder = new CASLoginResponse.CASLoginResponseBuilder(casMessageContext);
         String serviceUrlFromRequest = casMessageContext.getServiceURL();
         AuthenticationResult authnResult = processResponseFromFrameworkLogin(casMessageContext, identityRequest);
-        String acsURL = CASSSOUtil.getAcsUrl(serviceUrlFromRequest, casMessageContext.getRequest().getTenantDomain());
+        URL url = null;
+
+        try {
+            url = new URL(serviceUrlFromRequest);
+        } catch (MalformedURLException mfe) {
+            throw new FrameworkException("Error occurred while retrieving cas service url from: " + serviceUrlFromRequest, mfe);
+        }
+
+        String baseUrl = url.getProtocol() + "://" + url.getHost();
+        if (url.getPort() != -1) {
+            baseUrl = baseUrl + ":" + url.getPort();
+        }
+
+        if(log.isDebugEnabled()){
+            log.debug("Resolved base url for cas" + baseUrl);
+        }
+        String acsURL = CASSSOUtil.getAcsUrl(baseUrl, casMessageContext.getRequest().getTenantDomain());
+        if (StringUtils.equals(baseUrl, acsURL)) {
+            acsURL = serviceUrlFromRequest;
+        }
+
+        if(log.isDebugEnabled()){
+            log.info("Resolved acsURL for cas " + acsURL);
```

Comment:
Make this as log as debug log since it is inside a log.isDebugEnabled() check.
 log.info("") -> log.debug("")